src/server/browserApp.ts

```diff
@@ -15,31 +15,26 @@
  */
 
 import { ChildProcess, execSync } from 'child_process';
-import { ConnectOptions } from '../browser';
 import * as platform from '../platform';
 
 export class BrowserApp extends platform.EventEmitter {
   private _process: ChildProcess;
   private _gracefullyClose: () => Promise<void>;
-  private _connectOptions: ConnectOptions;
+  private _browserWSEndpoint: string | null = null;
 
-  constructor(process: ChildProcess, gracefullyClose: () => Promise<void>, connectOptions: ConnectOptions) {
+  constructor(process: ChildProcess, gracefullyClose: () => Promise<void>, wsEndpoint: string | null) {
     super();
     this._process = process;
     this._gracefullyClose = gracefullyClose;
-    this._connectOptions = connectOptions;
+    this._browserWSEndpoint = wsEndpoint;
   }
 
   process(): ChildProcess {
     return this._process;
   }
 
   wsEndpoint(): string | null {
-    return this._connectOptions.browserWSEndpoint || null;
-  }
-
-  connectOptions(): ConnectOptions {
-    return this._connectOptions;
+    return this._browserWSEndpoint;
```

Comment:
I didn't find a scenario where api returns a BrowserApp without a wsEndpoint. Let's change the signature?

Reply:
There are no external scenarios, but it is there inside the launch() method. Let's only say it is non-null on the api.md level.